packages/@aws-cdk/aws-appmesh/lib/http-route-path-match.ts

```diff
@@ -95,3 +95,123 @@ class HttpRouteWholePathMatch extends HttpRoutePathMatch {
+   *
+   * @default - rewrites the request's URL path to '/'
+   */
+  readonly prefixPathRewrite?: CfnGatewayRoute.HttpGatewayRoutePrefixRewriteProperty;
+}
+
+/**
+ * Defines HTTP gateway route matching based on the URL path of the request.
+ */
+export abstract class HttpGatewayRoutePathMatch {
+  /**
+   * The value of the path must match the specified value exactly.
+   * The provided `path` must start with the '/' character.
+   *
+   * @param path the exact path to match on
+   * @param exactPathRewrite the value to substitute for the matched part of the path of the gateway request URL
+   */
+  public static exactly(path: string, exactPathRewrite?: string): HttpGatewayRoutePathMatch {
+    return new HttpGatewayRouteWholePathMatch({ exact: path }, { exact: exactPathRewrite });
+  }
+
+  /**
+   * The value of the path must match the specified regex.
+   *
+   * @param regex the regex used to match the path
+   * @param exactPathRewrite the value to substitute for the matched part of the path of the gateway request URL
+   */
+  public static regex(regex: string, exactPathRewrite?: string): HttpGatewayRoutePathMatch {
+    return new HttpGatewayRouteWholePathMatch({ regex }, { exact: exactPathRewrite });
+  }
+
+  /**
+   * The value of the path must match the specified prefix.
+   *
+   * @param prefixPathMatch the value to use to match the beginning of the path part of the URL of the request.
+   *   When `rewriteTo` is provided, `prefixPathMatch` must start and end with the '/' character.
+   *   If provided as "/", matches all requests.
+   *   For example, if your virtual service name is "my-service.local"
+   *   and you want the route to match requests to "my-service.local/metrics", your prefix should be "/metrics".
+   * @param rewriteTo Specify either disabling automatic rewrite to '/' or rewriting to specified prefix path.
+   *  To disable automatic rewrite, provide `''`.
+   */
+  public static startsWith(prefixPathMatch: string, rewriteTo: string = '/'): HttpGatewayRoutePathMatch {
+    return new HttpGatewayRoutePrefixPathMatch(prefixPathMatch, rewriteTo);
+  }
+
+  /**
+   * Returns the gateway route path match configuration.
+   */
+  public abstract bind(scope: Construct): HttpGatewayRoutePathMatchConfig;
+}
+
+class HttpGatewayRoutePrefixPathMatch extends HttpGatewayRoutePathMatch {
+  constructor(
+    private readonly prefixPathMatch: string,
+    private readonly rewriteTo: string,
+  ) { super(); }
+
+  bind(_scope: Construct): HttpGatewayRoutePathMatchConfig {
+    return this.rewriteTo === ''
+      ? {
+        prefixPathMatch: this.prefixPathMatch,
+        prefixPathRewrite: {
+          defaultPrefix: 'DISABLED',
+        },
+      }
+      : this.rewriteTo === '/'
+        ? {
+          prefixPathMatch: this.prefixPathMatch,
+        }
+        : {
+          prefixPathMatch: this.prefixPathMatch,
+          prefixPathRewrite: {
+            value: this.rewriteTo,
+          },
+        };
```

Comment:
This is a pretty convoluted way to write this... notice that you're repeating `prefixPathMatch: this.prefixPathMatch` 3 times.

How about

```suggestion
     return {
        prefixPathMatch: this.prefixPathMatch,
        prefixPathRewrite: this.rewriteTo === '/' ? undefined : {
          defaultPrefix: this.rewriteTo === '' ? 'DISABLED' : undefined,
          value: this.rewriteTo === '' ? undefined : this.rewriteTo,
        },
      };
```

Reply:
This is definitely smarter way to implement it.